storm-client/src/jvm/org/apache/storm/Config.java

```diff
@@ -1731,21 +1744,28 @@
     @IsInteger
     @IsPositiveNumber
     public static final String WORKER_BLOB_UPDATE_POLL_INTERVAL_SECS = "worker.blob.update.poll.interval.secs";
+
     /**
-     * A specify Locale for daemon metrics reporter plugin. Use the specified IETF BCP 47 language tag string for a Locale.
+     * Specify the Locale for daemon metrics reporter plugin. Use the specified IETF BCP 47 language tag string for a Locale.
+     * This config should have been placed in the DaemonConfig class. Keeping it here only for backwards compatibility.
      */
     @IsString
     public static final String STORM_DAEMON_METRICS_REPORTER_PLUGIN_LOCALE = "storm.daemon.metrics.reporter.plugin.locale";
+
     /**
-     * A specify rate-unit in TimeUnit to specify reporting frequency for daemon metrics reporter plugin.
+     * Specify the rate unit in TimeUnit for daemon metrics reporter plugin.
+     * This config should have been placed in the DaemonConfig class. Keeping it here only for backwards compatibility.
      */
     @IsString
     public static final String STORM_DAEMON_METRICS_REPORTER_PLUGIN_RATE_UNIT = "storm.daemon.metrics.reporter.plugin.rate.unit";
+
     /**
-     * A specify duration-unit in TimeUnit to specify reporting window for daemon metrics reporter plugin.
+     * Specify the duration unit in TimeUnit for daemon metrics reporter plugin.
+     * This config should have been placed in the DaemonConfig class. Keeping it here only for backwards compatibility.
```

Comment:
Should these be deprecated if only for backwards compatibility?

Reply:
We need these configs. But they should sit in `DaemonConfig` class. I can't simply move it because if any topology invokes it directly, and if we move it and upgrade the cluster, I am afraid that this topology will have ClassNotFoundException. 

I thought about duplicating these configs in `DaemonConfig` and "deprecate" these in Config. I wasn't sure if it is a good idea since it adds two identical sets of configs.

Please let me know what your thoughts are

Reply:
I see.  Ideally I probably would duplicate and deprecate these for future removal or file a JIRA to do this another day.

Reply:
I can make the change

Reply:
I reverted the change because nimbus_test.clj is complaining.
```
Caused by: java.lang.IllegalStateException: STORM-DAEMON-METRICS-REPORTER-PLUGIN-DURATION-UNIT already refers to: #'org.apache.storm.daemon-config/STORM-DAEMON-METRICS-REPORTER-PLUGIN-DURATION-UNIT in namespace: org.apache.storm.nimbus-test
```

Will file a JIRA to deal with that once this PR is merged.